osf/models/preprint.py

```diff
@@ -580,15 +585,14 @@ def csl(self):  # formats node information into CSL format for citation parsing
 
     @property
     def is_latest_version(self):
-        if not self.is_published:
+        if not self.date_published:
             return False
         versioned_guid = self.versioned_guids.first()
         base_guid = versioned_guid.guid
         latest_version = base_guid.referent
-        if latest_version.is_published:
+        if latest_version.date_published:
             return latest_version.version == self.version
-        latest_version = Preprint.objects.filter(
-            is_published=True,
+        latest_version = Preprint.ever_published_objects.filter(
             id__in=base_guid.versions.values_list('object_id', flat=True)
         ).order_by('-versioned_guids__version').first()
         return latest_version.version == self.version if latest_version else False
```

Comment:
With `guid` always points to the latest version, can we simply check the following?

```python
if not self.date_published:
    return False
return self.versioned_guids.first().guid.referent.version == self.version
```